arangod/Network/Methods.cpp

```diff
@@ -268,7 +268,23 @@ class RequestsState final : public std::enable_shared_from_this<RequestsState> {
           tryAgainAfter = std::chrono::seconds(3);
         }
 
-        if ((now + tryAgainAfter) >= _endTime) { // cancel out
+        // Now check if the request was directed to an explicit server and see if that
+        // server is failed, if so, we should no longer retry, regardless of the timeout:
+        bool found = false;
+        if (_destination.size() > 7 && _destination.compare(0, 7, "server:") == 0) {
```

Comment:
shouldn't that be handled in retryLater?

Reply:
Well, I think the decision as to whether we do want to retry or not is taken here. Why not take it here and act immediately, instead of calling `retryLater` just to decide there to not retry later after all?